cmd/manager/main.go

```diff
@@ -114,14 +117,18 @@ func main() {
 	}
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:                 scheme,
-		MetricsBindAddress:     config.MetricsAddr,
-		Port:                   9443,
-		Namespace:              config.Namespace,
+		Scheme:  scheme,
+		Metrics: metricsserver.Options{BindAddress: config.MetricsAddr},
+		WebhookServer: webhook.NewServer(webhook.Options{
+			Port: 9443,
+		}),
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{config.Namespace: {}},
+			SyncPeriod:        &syncPeriod,
+		},
```

Comment:
Something is wrong here. The `users-iodc` e2e tests errors out with:
`unable to list: mongodb-atlas-system because of unknown namespace for the cache`

I suspect the `Namespace: config.Namespace`-> `Cache.Options.DefaultNamespaces` is wrong somehow, but cannot figure out how by reading their breaking changes notes here:
https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.16.0
or
https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0

Any hints welcome!
I think we are pretty close to getting this working, perhaps this is teh last issue to fix.

Reply:
Never mind, turns out for some reason the OIDC was passing a empty namespace to the created manager and it was working on version `0.15`, but that does not work on `0.16`. Being explicit about the namespace fixes the issue. I made that a separated commit for the review.